src/store/GitHubStore/GitHubStore.ts

```diff
@@ -0,0 +1,30 @@
+import ApiStore from 'shared/store/ApiStore';
+import {RequestParams, HTTPMethod} from 'shared/store/ApiStore/types';
+import {IGitHubStore, GetOrganizationReposListParams, ApiResp, SuccessRepoItem, ErrorRepoItem} from "./types";
+
+
+export default class GitHubStore implements IGitHubStore {
+    private readonly apiStore = new ApiStore('https://api.github.com'); // 
+
+    async getOrganizationReposList<RepoName = string>(params: GetOrganizationReposListParams): Promise<ApiResp<RepoName>> {
+        let result:ApiResp<RepoName> = {};
+
+        let paramsApi : RequestParams<any> = {
+            method: HTTPMethod.GET, //"GET",
+            endpoint:
+              this.apiStore.baseUrl + "/orgs/" + params.organizationName + "/repos",
+            headers: {},
+            data: {},
+          };
+    
+          let response = await this.apiStore.request<SuccessRepoItem<RepoName>, ErrorRepoItem>(paramsApi);
+          const {data, success} = response;
+          if (success) {
+            result.data = (data as SuccessRepoItem<RepoName>).map((item) => item.name);
```

Comment:
as возникает из-за того, что вы не должны тут деструктуризировать, ts не понимает, как data и success взаимосвязаны
```suggestion
            if (response.success) {
               return response.data.map...
            } else {
               return response.data.message
            }
```